### DISCUSSION TITLE
引用部の見出しも目次になる

### DISCUSSION THREAD

**Discussion by yoshi389111** [increments/qiita-discussions#441] [要望・アイデア・不具合]
Qiitaの記事を投稿すると目次が自動生成されると思うのですが、引用部にある見出し（ヘディング）も同じく目次扱いになっているようです。
これは意図した挙動なのでしょうか？
（別に大きな問題はないですが、ちょっと気になります）

例：下記のように書いた場合には、「引用した見出し」も目次に現れる

```
# 見出し１

> ### 引用した見出し

## 見出し２
```

具体例：[この記事](https://qiita.com/yoshi389111/items/b8b6e679a1c4cbd1fe0f) だと、GitHub の blog の内容を、見出しも含めて引用しているのですが、目次にもそれが出てきています。

**Comment by yarakos95** [comment]
補足なのですが、この問題はブロッククォートに限らず、他の Container blocks でも生じます。

Qiita で対応している Container blocks は以下の通りです：
- ブロッククォート
- 箇条書きリスト
- 番号付きリスト
- タスクリスト
- 脚注
- Note

実際に試してみると、この 5 つの中に見出しをネストした際に目次として認識されています。

![headingsINcontainer](https://user-images.githubusercontent.com/75878492/222259648-794b0f8c-b62c-4ee0-8563-c34a11139678.png)

<details><summary>サンプルコード</summary>

```markdown
> # ブロッククォート内の見出し


- # 箇条書きリスト内の見出し
- 項目



1. # 番号付きリスト内の見出し
1. 項目



- [ ] 項目
    # タスクリスト内の見出し
- [x] 項目



本文本文[^footnote]

[^footnote]: # 脚注内の見出し

:::note info
# Note 内の見出し
:::
```

https://qiita.com/Yarakashi_Kikohshi/private/1ce1bcfe91dae91785cd

</details>

実際問題として、ブロッククォート以外で見出しをネストするシーンがあるのかは疑問が残りますが、ブロッククォート内の見出しを目次に含めないならば、ほかの Container blocks 内の見出しも目次に入るべきではないと思います。

**Comment by kyntk** [comment]
@yoshi389111
ご意見ありがとうございます！
おっしゃるとおり、現在は引用部分の見出しも目次に含まれる実装になっています。
また、GitHub の Wiki でも同様に、引用部分の見出しも目次に含まれるようです。(例: https://github.com/wataru86/wiki-sample/wiki)

引用部分の見出しが目次に含まれることに対して、違和感があるというのも分かるので、他の開発優先度なども踏まえて修正を検討します :bow:

@Yarakashi-Kikohshi
補足していただきありがとうございます :sparkles:

**Reply by yoshi389111** [reply]
了解です。

回答、ありがとうございました。
